src/algebra/homology/local_cohomology.lean

```diff
@@ -87,17 +86,33 @@ of `R`, represented as a functor `I`. -/
 In this definition we do not assume any special property of the diagram `I`, but the relevant case
 will be where `I` is (cofinal with) the diagram of powers of a single given ideal.
 
-Below, we give two equivalent (to be shown) definitions of the usual local cohomology with support
+Below, we give two equivalent definitions of the usual local cohomology with support
 in an ideal `J`, `local_cohomology` and `local_cohomology.of_self_le_radical`.
 
-TODO: Show that any functor cofinal with `I` gives the same result.
  -/
 def of_diagram (I : D ⥤ ideal R) (i : ℕ) :
   Module.{max u v} R ⥤ Module.{max u v} R :=
 colimit (diagram.{(max u v) v} I i)
 
 end
 
+section
+variables {R : Type max u v v'} [comm_ring R] {D : Type v} [small_category D]
+
+variables {E : Type v'} [small_category E]
+  (I' : E ⥤ D) (I : D ⥤ ideal R)
+
+/-- Local cohomology along a composition of diagrams. -/
+def diagram_comp (i : ℕ) : diagram (I' ⋙ I) i ≅ I'.op ⋙ (diagram I i) := iso.refl _
+
+/-- Local cohomology agrees along arbitrary cofinal diagrams. -/
```

Comment:
There are several layers of abstraction here, so the answer is both yes and no. The `functor.final` class is a "final inclusion of one diagram into another". So you could construct the inclusion `F` of the diagram `D2` of powers of I^2 into the diagram `D3` of powers of I^3, and show that it is final. Then you will get that `F ⋙ D2` and `D3` give the same local cohomology. But this is still formally a different object than the colimit of `D2` itself.

And, this is distinct from the approach you might be expecting -- where you show that two diagrams of ideals are "cofinal with respect to each other" within the poset (category) of all ideals. This definition simply doesn't seem to exist in mathlib.

Reply:
Hmm, maybe it would be better to say that this shows agreement along an arbitrary final inclusion of one diagram into another. To show the thing you said, you would probably chain together two isomorphisms given by including the powers of I^2 into all powers of I, and likewise for I^3.

Reply:
yeah I would definitely be happier with a docstring of that nature.

Reply:
Sounds good. One technical challenge here is that an isomorphism is data and the human statement "local cohomology computed via powers of I^2 and I^3 agrees" could mean any number of things -- isomorphisms from comparing various diagrams. The resulting isomorphisms are all equal (of course), but not definitionally so.

So for Lean's purposes I remain unsure what is really the best way to proceed here. This was part of the motivation for defining the `self_le_radical` diagram elsewhere in the file: it's in some sense the unique largest diagram to consider, so a natural one for comparing everything else to.

Reply:
Would there be any substantive benefit to stating and proving that two cofinal diagrams give rise to the same local cohomology by using your theorem twice?

Reply:
Also:  Kevin, as I get a chance to work more on Koszul complexes, should I shift to Lean 4?

Reply:
Emily, that should be doable and might be a useful variant, yes. Right now the file has `iso_of_same_radical` down at the bottom, which chains together three isomorphisms (comparing the powers of I and J to their respective `self_le_radical` diagrams, plus equating the two large diagrams).

Reply:
> Would there be any substantive benefit to stating and proving that two cofinal diagrams give rise to the same local cohomology by using your theorem twice?

I attempted this and while it looks doable, I think it should go in a separate PR since (surprise surprise) it seems lengthy to do.